cmd/wego/flux/cmd.go

```diff
@@ -0,0 +1,63 @@
+package flux
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/BurntSushi/toml"
+	"github.com/spf13/cobra"
+)
+
+//go:embed bin/flux
+var fluxExe []byte
+
+type dependencies struct {
+	Flux flux
+}
+type flux struct {
+	Version string
+}
+
+var Cmd = &cobra.Command{
+	Use:   "flux",
+	Short: "Use flux commands",
+	Run:   runCmd,
+}
+var path = "temp/bin"
+var exePath string
+
+func init() {
+	var dep dependencies
+	if _, err := toml.DecodeFile("tools/dependencies.toml", &dep); err != nil {
```

Comment:
This won't work. This runs every time you run your binary, not at build time, which means if you run the `wego` binary outside the wego repo it breaks. 

```
weave-gitops git:(flux-bin) cd
➜  ~ /Users/luiz/Projects/weaveworks/weave-gitops/bin/wego flux 
Error: open tools/dependencies.toml: no such file or directory
```

you should use build flags instead, you can take a look at the version command. You can then use the `stoml` tool to read the version from the deps file, look at how to use that in the `tools/download-deps.sh` file.

Regarding the path you save the file, we should save it under `$HOME/.wego/bin` and use it from there, if you keep this way saving in `temp/bin` it will save the file in the user's current folder wherever they are running the command.

I got a bug, the way you passed the rest of the arguments to the flux bin don't work.

```
weave-gitops git:(flux-bin) bin/wego flux install -h
Use flux commands

Usage:
  wego flux [flags]

Flags:
  -h, --help   help for flux

Global Flags:
  -v, --verbose   Enable verbose output
```
when running that, we should see flux output, something like:

```
install -h
The install command deploys Flux in the specified namespace.
If a previous version is installed, then an in-place upgrade will be performed.

Usage:
  flux install [flags]

Examples:
  # Install the latest version in the flux-system namespace
  flux install --version=latest --namespace=flux-system
....
```

It looks like cobra takes control over everything and thinks those arguments are to it. I'm not sure how to solve that one, you'll have to google that one, let me know if you get stuck with that, I can help you.

Reply:
So i gotta ask.  At what point does embedding this not make sense anymore. I mean if it's listed as a dependency and getting installed can we not just use the installed flux bin instead of the embedded one.

Reply:
Those dependencies are build dependencies. They are gonna be installed only on our machines and in the ci. and that installing process is done via make file, so most of those are only there to assist us during development. That's why we embed flux and copy to user's file system, otherwise, we would need to ask them to manually install flux before using wego.

Reply:
The only down side I see to that is that the main reason for being for wego is to make the process of using flux2 simple and intuitive. Perhaps the wego binary could do a download and install if it doesn't find flux locally?

Reply:
maybe use `toml.Decode` instead of `toml.DecodeFile` so you can avoid file path location issue 🙂

Reply:
I was able to avoid those issues. Using the change above and running the command like this `./weave-gitops/bin/wego flux -- install -h` . Not 100% we want that. 

<img width="603" alt="image" src="https://user-images.githubusercontent.com/3892786/113909661-8ca1b680-979d-11eb-9de8-9eb703dc57af.png">

Reply:
I mean its not great but its better than what i had.

Reply:
since this is only for debugging, I'm good with that. Great discovery.